### PR TITLE
[Issue 5618] [docs]Correct python example

### LINKED ISSUE #5618: [Docs]The document of python function example is incorrect
**Describe the bug**
There was an error in the example of using [python function](http://pulsar.apache.org/docs/en/functions-develop/#example) that causes the following exception to be thrown:

```
2019-11-11 09:48:43 +0000] [ERROR] log.py: Traceback (most recent call last):
[2019-11-11 09:48:43 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/python_instance_main.py", line 211, in <module>
[2019-11-11 09:48:43 +0000] [ERROR] log.py: main()
[2019-11-11 09:48:43 +0000] [ERROR] log.py:   File "/pulsar/instances/python-[instance/python_instance_main.py", line 192, in main
[2019-11-11 09:48:43 +0000] [ERROR] log.py: pyinstance.run()
[2019-11-11 09:48:43 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/python_instance.py", line 186, in run
[2019-11-11 09:48:43 +0000] [ERROR] log.py: self.input_serdes[topic] = serde_kclass()
[2019-11-11 09:48:43 +0000] [ERROR] log.py: TypeError
[2019-11-11 09:48:43 +0000] [ERROR] log.py: :
[2019-11-11 09:48:43 +0000] [ERROR] log.py: __init__() takes exactly 2 arguments (1 given)
```

Parameters should not appear in the initialization function of class TweetSerDe, messages should come from the topic and should not be assigned during initialization. the document example is wrong, and the correct example should be:

```
class Tweet(object):
    def __init__(self, username, tweet_content):
        self.username = username
        self.tweet_content = tweet_content


from pulsar import SerDe

class TweetSerDe(SerDe):
    def __init__(self):
        pass

    def serialize(self, input):
        return bytes("{0}|{1}".format(input.username, input.tweet_content))

    def deserialize(self, input_bytes):
        tweet_components = str(input_bytes).split('|')
        return Tweet(tweet_components[0], tweet_componentsp[1])
```

Reference: https://github.com/apache/pulsar/blob/master/pulsar-functions/python-examples/custom_object_function.py

**Expected behavior**
Fix document

**Additional context**
Use the correct example on the document

### PR DESCRIPTION
**Pull request by Jennifer88huang**
Fixes #5618 

### Motivation

The python example is not correct.

### Modifications

1. Correct the python example.
2. Correct two link issues.

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
@tuteng @candlerb Shall we remove 
```

File: site2/docs/functions-develop.md [hunk 1]
```diff
@@ -235,20 +234,20 @@ In order to use this class in Pulsar Functions, you have two options:
...
   ```python
-  from pulsar import SerDe
+from pulsar import SerDe
 
-  class TweetSerDe(SerDe):
-      def __init__(self, tweet):
-          self.tweet = tweet
+class TweetSerDe(SerDe):
+    def __init__(self):
+        pass
```
Comment (merlimat) [inline site2/docs/functions-develop.md]: The default constructor can be removed

Comment (Jennifer88huang) [reply site2/docs/functions-develop.md]: Got it, thanks for your confirmation.

**Comment by tuteng** [comment]
This contents of the file https://github.com/apache/pulsar/blob/master/site2/website/versioned_docs/version-2.4.1/functions-develop.md also seem to need updating.

**Comment by Jennifer88huang** [comment]
@tuteng Thanks for your reminder. If the content in this PR is correct, i'll update the one in version 2.4.1.

**Comment by Jennifer88huang** [comment]
run java8 tests

**Pull request merged**